solr/contrib/prometheus-exporter/src/java/org/apache/solr/prometheus/exporter/MetricsConfiguration.java

```diff
@@ -77,22 +79,22 @@ public PrometheusExporterSettings getSettings() {
     return searchConfiguration;
   }
 
-  public static MetricsConfiguration from(String path) throws Exception {
+  public static MetricsConfiguration from(String resource) throws Exception {
     // See solr-core XmlConfigFile
     final DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
     try {
       dbf.setXIncludeAware(true);
       dbf.setNamespaceAware(true);
     } catch (UnsupportedOperationException e) {
-      log.warn("{} XML parser doesn't support XInclude option", path);
+      log.warn("{} XML parser doesn't support XInclude option", resource);
     }
 
     Document document;
-    File file = new File(path);
-    if (file.isFile()) {
-      document = dbf.newDocumentBuilder().parse(file);
+    Path path = Path.of(resource);
+    if (Files.exists(path)) {
+      document = dbf.newDocumentBuilder().parse(path.toAbsolutePath().toString());
```

Comment:
This `parse(String)` is documented to take a URI.  I don't think a file path is a URI?  If you look at the implementation of the variant that takes a File, it converts it to a URI then and then an ASCII string.  This could be done via `path.toUri().toASCIIString()`.   FWIW I'd also be okay with just doing path.toFile() here.

Reply:
Good catch. I tested it in multiple configurations, but there's probably some edge case it doesn't work for. I'll use `path.toUri().toASCIIString()` since the `parse(file)` method just gets the URI of the file and passes that to the `parse(uri)` implementation anyways. Might as well skip the middleman-method.